dao/src/main/resources/db/migration/V86__ortRunDeleteCascade.sql

```diff
@@ -0,0 +1,282 @@
+-- This migration script modifies the foreign key constraints of the tables that reference ort_runs to add an
+-- ON DELETE CASCADE clause. With this setup, it is very easy to delete an ORT run and all its associated data.
+
+ALTER TABLE notifier_runs
+  DROP CONSTRAINT notifier_runs_notifier_job_id_fkey,
+  ADD CONSTRAINT notifier_runs_notifier_job_id_fkey
+    FOREIGN KEY (notifier_job_id) REFERENCES notifier_jobs (id) ON DELETE CASCADE;
+
+ALTER TABLE notifier_jobs
+  DROP CONSTRAINT notifier_jobs_ort_run_id_fkey,
+  ADD CONSTRAINT notifier_jobs_ort_run_id_fkey
+    FOREIGN KEY (ort_run_id) REFERENCES ort_runs (id) ON DELETE CASCADE;
+
+ALTER TABLE packages_analyzer_runs
+  DROP CONSTRAINT packages_analyzer_runs_analyzer_run_id_fkey,
+  ADD CONSTRAINT packages_analyzer_runs_analyzer_run_id_fkey
+    FOREIGN KEY (analyzer_run_id) REFERENCES analyzer_runs (id) ON DELETE CASCADE;
```

Comment:
IIUC `ON DELETE CASCADE` only partly works for many to many associations. In this example it will remove the row from the `packages_analyzer_runs` join table if the analyzer run is deleted, but it will not remove the package itself. So some other mechanism is needed to clean up dangling packages which are not associated to any analyzer runs anymore.
This is not necessarily a problem, but I think it should be pointed out in the comment on top of the file and the commit message.

Reply:
One good mechanism for that would be an after delete trigger like [this](https://stackoverflow.com/a/35373676).

Reply:
Please, be careful with all kind of triggers. They can decrease performance dramatically.

Reply:
> To deal with possible orphan entries, I was thinking about a periodic task that does some cleanup.

I think I would prefer that over a trigger, having this in the application code is more accessible to the developers and also we might need more control over in which cases we want to remove orphaned entries.